azurerm/resource_arm_cosmos_db_account.go

```diff
@@ -205,6 +206,26 @@ func resourceArmCosmosDBAccount() *schema.Resource {
 				Set: resourceAzureRMCosmosDBAccountCapabilitiesHash,
 			},
 
+			"is_virtual_network_filter_enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"virtual_network_rule": {
```

Comment:
Would this make more sense as a simple set of strings `virtual_network_rule_subnet_ids` rather then a set of blocks containing id?

```
virtual_network_rule_subnet_ids  =[
"${azurerm_subnet.subnet1.id}",
"${azurerm_subnet.subnet2.id}",
]
```

Reply:
I think that we should err on the side of leaving this as-is: if the Azure API adds additional fields to these rules in the future, having a resource here rather than a simple ID makes such a change easier to accommodate. This would also align better with a future `cosmosdb_account_virtual_network_rule` resource as suggested in #1342.

Reply:
Good points 👍